programs/uxd/src/instructions/rebalance_alloyx_vault_depository.rs

```diff
@@ -0,0 +1,773 @@
+        let liquidity_collateral_amount_before = self.alloyx_vault_collateral.amount;
+        let outstanding_collateral_amount_before = self.alloyx_vault_info.wallet_desk_usdc_value;
+        let total_shares_supply_before = self.alloyx_vault_mint.supply;
+        let total_shares_value_before = checked_add(
+            liquidity_collateral_amount_before,
+            outstanding_collateral_amount_before,
+        )?;
+        let owned_shares_amount_before = self.alloyx_vault_depository_shares.amount;
+        let owned_shares_value_before = compute_value_for_shares_amount_floor(
+            owned_shares_amount_before,
+            total_shares_supply_before,
+            total_shares_value_before,
+        )?;
+
+        // Compute withdrawn amount
+        let collateral_amount_before_precision_loss = std::cmp::min(
+            desired_collateral_amount,
+            liquidity_collateral_amount_before,
+        );
+        let shares_amount = compute_shares_amount_for_value_floor(
+            collateral_amount_before_precision_loss,
+            total_shares_supply_before,
+            total_shares_value_before,
+        )?;
+        if shares_amount == 0 {
+            return Ok(0);
+        }
+        let collateral_amount_after_precision_loss = compute_value_for_shares_amount_floor(
+            shares_amount,
+            total_shares_supply_before,
+            total_shares_value_before,
+        )?;
+        if collateral_amount_after_precision_loss == 0 {
+            return Ok(0);
+        }
+        let collateral_amount_delta_precision_loss = checked_sub(
+            collateral_amount_before_precision_loss,
+            collateral_amount_after_precision_loss,
+        )?;
+
+        // Actually runs the CPI
+        let alloyx_vault_info = self.alloyx_vault_depository.load()?.alloyx_vault_info;
+        let collateral_mint = self.alloyx_vault_depository.load()?.collateral_mint;
+        let alloyx_vault_depository_pda_signer: &[&[&[u8]]] = &[&[
+            ALLOYX_VAULT_DEPOSITORY_NAMESPACE,
+            alloyx_vault_info.as_ref(),
+            collateral_mint.as_ref(),
+            &[self.alloyx_vault_depository.load()?.bump],
+        ]];
+        alloyx_cpi::cpi::withdraw(
+            self.into_withdraw_from_alloyx_vault_to_alloyx_vault_depository_context()
+                .with_signer(alloyx_vault_depository_pda_signer),
+            vault_id.to_owned(),
+            shares_amount,
+        )?;
+        token::transfer(
+            self.into_transfer_payer_collateral_to_identity_depository_collateral_context(),
+            collateral_amount_delta_precision_loss,
+        )?;
+        token::transfer(
+            self.into_transfer_alloyx_vault_depository_collateral_to_identity_depository_collateral_context()
+                .with_signer(alloyx_vault_depository_pda_signer),
+            collateral_amount_after_precision_loss,
+        )?;
+
+        // Reload onchain data
```

Comment:
I don't think we ever read `alloyx_vault_shares` in our code, it is strictly used by the alloyx program, I don't think that's necessary ? I will try out if it makes any difference but I think we don't need that, since we're never "load()"'ing the content of that account

Reply:
update: tried with and without         `self.alloyx_vault_shares.reload()?;`
, didn't see any functional difference 🤔